assets/src/edit-story/components/panels/size.js

```diff
@@ -18,101 +18,218 @@
 import { dataPixels } from '../../units';
+import { ReactComponent as Locked } from '../../icons/lock.svg';
+import { ReactComponent as Unlocked } from '../../icons/unlock.svg';
+import { ReactComponent as Fullbleed } from '../../icons/fullbleed.svg';
+import { ReactComponent as FlipVertical } from '../../icons/flip_vertical.svg';
+import { ReactComponent as FlipHorizontal } from '../../icons/flip_horizontal.svg';
+import Toggle from '../form/toggle';
 import { SimplePanel } from './panel';
 import getCommonValue from './utils/getCommonValue';
 
+const ExpandedNumeric = styled(Numeric)`
+  flex: 1;
+`;
+
+const BoxedNumeric = styled(ExpandedNumeric)`
+  padding: 6px 6px;
+  border: 1px solid ${({ theme }) => theme.colors.fg.v3};
+  border-radius: 4px;
+`;
+
+const FlipButton = styled(Button)`
+  border: none;
+  padding: 0;
+`;
+
+const FlipButtonIcon = styled.svg`
+  opacity: 0.54;
+`;
+
 function SizePanel({ selectedElements, onSetProperties }) {
+  const x = getCommonValue(selectedElements, 'x');
+  const y = getCommonValue(selectedElements, 'y');
   const width = getCommonValue(selectedElements, 'width');
   const height = getCommonValue(selectedElements, 'height');
   const isFill = getCommonValue(selectedElements, 'isFill');
-  const [state, setState] = useState({ width, height });
+  const rotationAngle = getCommonValue(selectedElements, 'rotationAngle');
+  const [state, setState] = useState({
+    x,
+    y,
+    width,
+    height,
+    isFill,
+    rotationAngle,
+  });
   const [lockRatio, setLockRatio] = useState(true);
   useEffect(() => {
-    setState({ width, height });
-  }, [width, height]);
-  const handleSubmit = (evt) => {
-    onSetProperties(({ width: oldWidth, height: oldHeight }) => {
-      let { width: newWidth, height: newHeight } = state;
-      if (lockRatio && (newHeight === '' || newWidth === '')) {
-        const ratio = oldWidth / oldHeight;
-        if (newWidth === '') {
-          newWidth = dataPixels(newHeight * ratio);
-        } else {
-          newHeight = dataPixels(newWidth / ratio);
+    setState({ x, y, width, height, isFill, rotationAngle });
+  }, [x, y, width, height, isFill, rotationAngle]);
+  useEffect(() => {
+    updateProperties();
+  }, [state.isFill, updateProperties]);
+  const updateProperties = useCallback(
+    (evt) => {
+      onSetProperties(({ width: oldWidth, height: oldHeight }) => {
+        const { height: newHeight, width: newWidth, ...rest } = state;
+        const update = { width: newWidth, height: newHeight, ...rest };
+        const hasHeightButNotWidth = newHeight === '' || newWidth === '';
+        const hasWidthButNotHeight = newHeight === '' && newWidth === '';
+
+        if (lockRatio && hasHeightButNotWidth && !hasWidthButNotHeight) {
+          const ratio = oldWidth / oldHeight;
+          if (newWidth === '') {
+            update.width = dataPixels(newHeight * ratio);
+          } else if (newHeight === '') {
```

Comment:
And the condition here would not be needed, as the implication from all the above would be, that `hasWidthButNotHeight` has to be `true` here.

Reply:
We don't need else if. Just else needed.